.github/workflows/integration-test.yml

```
@@ -105,3 +104,41 @@ jobs:
         # For every other version and/or OS, run pytest without coverage
         if: (matrix.os != 'ubuntu-latest') || (matrix.python-version != env.TARGET_PYTHON_VERSION )
         run: poetry run pytest ${{ github.event.inputs.pytest_addopts }}
+  build:
+    name: Build versioned package
+    # This job is used to build the package and upload the artifacts
```

Consider adding an explicit mention of the upload target to the comment here. It might also be useful to link to this anchor (which has a helpful screenshot): https://github.com/actions/upload-artifact?tab=readme-ov-file#where-does-the-upload-go

```suggestion
    # This job is used to build the package and upload the artifacts to GitHub Actions workflow results.
```

---

Done.